src/python_framework_v02/troute/nhd_io.py

```diff
@@ -196,18 +202,25 @@ def get_ql_from_wrf_hydro_mf(qlat_files, index_col="feature_id", value_col="q_la
         preprocess=drop_all_coords,
         # parallel=True,
     ) as ds:
+        glob_step = 1
+        if not file_run_size:
+            glob_slice = slice(None, None, glob_step)
+        else:
+            glob_slice_idx_0 = int(ts_iterator * file_run_size)
+            glob_slice_idx_1 = int(ts_iterator + 1) * file_run_size
+            glob_slice = slice(glob_slice_idx_0, glob_slice_idx_1, glob_step)
```

Comment:
I recommend taking this file-level manipulation back out of this function and leaving it so that the function reads whatever is handed to it -- any reduction would happen on the control side and fewer files would be handed over in the list.